compiler/src/dotty/tools/dotc/config/Feature.scala

```diff
@@ -121,7 +121,8 @@ object Feature:
 
   def namedTypeArgsEnabled(using Context) = enabled(namedTypeArguments)
 
-  def clauseInterleavingEnabled(using Context) = enabled(clauseInterleaving)
+  def clauseInterleavingEnabled(using Context) =
+    sourceVersion.isAtLeast(`3.6`) || enabled(clauseInterleaving)
```

Comment:
this shouldn't be mixing source version with language import checks, they should be kept separate,
e.g. in the parser do `sourceVersion.isAtLeast(`3.6`) || in.featureEnabled(Feature.clauseInterleaving)`

Reply:
not sure to see why we should do that ?

Reply:
ok, I guess its fine then, curious that in parser they just have `Feature.fewerBracesEnabled` without explicit using

Reply:
Yea, I was currious too. I guess it might be a no-op as it is, therefore a bug ?